packages/grid/x-data-grid/src/components/base/GridBody.tsx

```diff
@@ -57,14 +57,27 @@ function GridBody(props: GridBodyProps) {
 
   const columnHeadersRef = React.useRef<HTMLDivElement>(null);
   const columnsContainerRef = React.useRef<HTMLDivElement>(null);
-  const windowRef = React.useRef<HTMLDivElement>(null);
   const renderingZoneRef = React.useRef<HTMLDivElement>(null);
 
   apiRef.current.columnHeadersContainerElementRef = columnsContainerRef;
   apiRef.current.columnHeadersElementRef = columnHeadersRef;
-  apiRef.current.windowRef = windowRef; // TODO rename, it's not attached to the window anymore
   apiRef.current.renderingZoneRef = renderingZoneRef; // TODO remove, nobody should have access to internal parts of the virtualization
 
+  const targetRef = React.useCallback(
+    (node: HTMLDivElement) => {
+      if (node && apiRef.current.windowRef?.current == null) {
+        apiRef.current.windowRef = { current: node };
+        const unsubscribe = apiRef.current.subscribeEvent('stateChange', () => {
+          setTimeout(() => {
+            apiRef.current.publishEvent('gridReady');
```

Comment:
It doesn't seem like it fixes the issue: https://codesandbox.io/s/confident-forest-6tizwv?file=/src/App.tsx

Reply:
does it not scroll to the index for you?
![image](https://user-images.githubusercontent.com/34584266/199308307-e7640353-751d-4d3a-b6b3-a09799a4d277.png)

I add `setTimeout` because some states like `rowMeta` might not finish initializing after row data is set. I am experimenting with a way to check if all the states have finished before `gridReady` is triggered.
```javascript
const unsubscribe = apiRef.current.subscribeEvent('stateChange', (newState) => { 
     if (newState.rowsMeta.currentPageTotalHeight > 0) {
          setTimeout(() => {
              apiRef.current.publishEvent('gridReady');
          }, 0);
          unsubscribe();
          }
        });
}
```